sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// A <see cref="JsonObjectSerializer"/> implementation that uses <see cref="JsonSerializer"/> to for serialization/deserialization.
+    /// </summary>
+    public class JsonObjectSerializer : ObjectSerializer
+    {
+        private readonly JsonSerializerOptions _options;
+
+        /// <summary>
+        /// Initializes new instance of <see cref="JsonObjectSerializer"/>.
+        /// </summary>
+        public JsonObjectSerializer() : this(new JsonSerializerOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes new instance of <see cref="JsonObjectSerializer"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> instance to use when serializing/deserializing.</param>
+        public JsonObjectSerializer(JsonSerializerOptions options)
+        {
+            _options = options;
+        }
+
+        /// <inheritdoc />
+        public override void Serialize(Stream stream, object? value, Type inputType)
+        {
+            var buffer = Encoding.UTF8.GetBytes(JsonSerializer.Serialize(value, inputType, _options));
```

Comment:
Should we use https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serializetoutf8bytes?view=netcore-3.1 instead?

Reply:
Yup.